From 9f6312e0c2504c27b72bb228a37ed5ac58776e94 Mon Sep 17 00:00:00 2001
From: Brian Dolbec <dolsen@gentoo.org>
Date: Wed, 10 Jun 2020 07:31:45 -0700
Subject: [PATCH] Fix _test_visualize.py twisted import errors

When graphviz modules are installed but not twisted, this creates test failures
in _discover.py which hard depends on twisted being installed.

Signed-off-by: Brian Dolbec <dolsen@gentoo.org>
---
 automat/_test/test_visualize.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/automat/_test/test_visualize.py b/automat/_test/test_visualize.py
index 987eb3c..142c4bf 100644
--- a/automat/_test/test_visualize.py
+++ b/automat/_test/test_visualize.py
@@ -62,6 +62,7 @@ def sampleMachine():
 
 
 @skipIf(not isGraphvizModuleInstalled(), "Graphviz module is not installed.")
+@skipIf(not isTwistedInstalled(), "Twisted is not installed.")
 class ElementMakerTests(TestCase):
     """
     L{elementMaker} generates HTML representing the specified element.
@@ -134,6 +135,7 @@ def isLeaf(element):
 
 
 @skipIf(not isGraphvizModuleInstalled(), "Graphviz module is not installed.")
+@skipIf(not isTwistedInstalled(), "Twisted is not installed.")
 class TableMakerTests(TestCase):
     """
     Tests that ensure L{tableMaker} generates HTML tables usable as
@@ -214,6 +216,7 @@ class TableMakerTests(TestCase):
 
 @skipIf(not isGraphvizModuleInstalled(), "Graphviz module is not installed.")
 @skipIf(not isGraphvizInstalled(), "Graphviz tools are not installed.")
+@skipIf(not isTwistedInstalled(), "Twisted is not installed.")
 class IntegrationTests(TestCase):
     """
     Tests which make sure Graphviz can understand the output produced by
@@ -232,6 +235,7 @@ class IntegrationTests(TestCase):
 
 
 @skipIf(not isGraphvizModuleInstalled(), "Graphviz module is not installed.")
+@skipIf(not isTwistedInstalled(), "Twisted is not installed.")
 class SpotChecks(TestCase):
     """
     Tests to make sure that the output contains salient features of the machine
--
libgit2 0.99.0

